### ISSUE TITLE
401 failed to resolve with github dependency

### ISSUE THREAD

**Issue by Mara-Li** [oven-sh/bun#14557]
### What version of Bun is running?

1.1.30+7996d06b8

### What platform is your computer?

Microsoft Windows NT 10.0.22631.0 x64

### What steps can reproduce the bug?

1. `bun init`
2. Add the following package in `devDependencies`:
```
{
  "name": "foo",
  "module": "index.js",
  "type": "module",
  "devDependencies": {
    "@types/bun": "latest",
    "@codemirror/language" : "https://github.com/lishid/cm-language"
  },
  "peerDependencies": {
    "typescript": "^5.0.0"
  }
}
```
3. Run `bun i`

### What is the expected behavior?

Installation from the package without issue

### What do you see instead?

 ```
error: GET https://api.github.com/repos/lishid/cm-language/tarball/ - 401
error: @codemirror/language@https://github.com/lishid/cm-language failed to resolve
```

### Additional information

No error happens when using `pnpm`

**Comment by capaj** [comment]
this happens on all OSes

**Comment by woaitsAryan** [comment]
+1

**Comment by frytg** [comment]
btw, `git@github.com:example/example.git#some-hash-or-branch` seems to work vs. `example/example#some-hash-or-branch`

**Comment by Jarred-Sumner** [comment]
We will have a fix for this in Bun v1.1.33

**Comment by RiskyMH** [comment]
Bun  `v1.1.33` is now released! https://bun.sh/blog/bun-v1.1.33
